src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -0,0 +1,608 @@
+     */
+    public boolean isJumpable() {
+        return isSubclass(JumpableGenerator.class);
+    }
+
+    /**
+     * Return true if random generator is jumpable and can leap to a very distant
+     * point in the state cycle.
+     *
+     * @return true if random generator is leapable.
+     */
+    public boolean isLeapable() {
+        return isSubclass(LeapableGenerator.class);
+    }
+
+    /**
+     * Return true if random generator can be cloned into a separate object with
+     * the same properties but positioned further in the state cycle.
+     *
+     * @return true if random generator is splittable.
+     */
+    public boolean isSplittable() {
+        return isSubclass(SplittableGenerator.class);
+    }
+
+    /**
+     * Return true if random generator can be used to create
+     * {@link java.util.stream.Stream Streams} of random numbers.
+     *
+     * @return true if random generator is streamable.
+     */
+    public boolean isStreamable() {
+        return isSubclass(StreamableGenerator.class);
+    }
+
+    /**
+     * Create an instance of {@link RandomGenerator} based on
+     * <a href="package-summary.html#algorithms">algorithm</a> chosen.
+     *
+     * @return new in instance of {@link RandomGenerator}.
+     *
+     */
+    public T create() {
+        try {
+            ensureConstructors();
+            return ctor.newInstance();
+        } catch (Exception ex) {
+            // Should never happen.
+            throw new IllegalStateException("Random algorithm " + name() + " is missing a default constructor", ex);
+        }
+    }
+
+    /**
+     * Create an instance of {@link RandomGenerator} based on
+     * <a href="package-summary.html#algorithms">algorithm</a> chosen
+     * providing a starting long seed. If long seed is not supported by an
+     * algorithm then the no argument form of create is used.
+     *
+     * @param seed long random seed value.
+     *
+     * @return new in instance of {@link RandomGenerator}.
+     */
+    public T create(long seed) {
+        try {
+            ensureConstructors();
+            return ctorLong.newInstance(seed);
+        } catch (Exception ex) {
+            return create();
+        }
+    }
+
+    /**
+     * Create an instance of {@link RandomGenerator} based on
+     * <a href="package-summary.html#algorithms">algorithm</a> chosen
+     * providing a starting byte[] seed. If byte[] seed is not supported by an
+     * <a href="package-summary.html#algorithms">algorithm</a> then the no
+     * argument form of create is used.
+     *
+     * @param seed byte array random seed value.
+     *
+     * @return new in instance of {@link RandomGenerator}.
+     *
+     * @throws NullPointerException if seed is null.
+     */
+    public T create(byte[] seed) {
+        Objects.requireNonNull(seed, "seed must not be null");
+        try {
+            ensureConstructors();
+            return ctorBytes.newInstance((Object)seed);
```

Comment:
IntelliJ warns that the cast to (Object) is redundant.

Reply:
Modified